File: CppAnswer/Q1.cpp
#include <iostream>
#include <vector>
#include <list>
#include <utility>

class HashMap {
private:
    std::vector<std::list<std::pair<int, int>>> map;
    int size;
    int hash(int key) {
        return key % size;
    }
    
public:
    HashMap(int s = 100) : size(s) {
        map.resize(size);
    }
    
    void put(int key, int value) {
        int hashValue = hash(key);
        for (auto& pair : map[hashValue]) {
            if (pair.first == key) {
                pair.second = value;
                return;
            }
        }
        map[hashValue].emplace_back(key, value);
    }
    
    int get(int key) {
        int hashValue = hash(key);
        for (const auto& pair : map[hashValue]) {
            if (pair.first == key) {
                return pair.second;
            }
        }
        return -1;
    }
    
    void remove(int key) {
        int hashValue = hash(key);
        map[hashValue].remove_if([key](const std::pair<int, int>& pair) {
            return pair.first == key;
        });
    }
};

int main() {
    HashMap hashMap;
    hashMap.put(1, 10);
    hashMap.put(2, 20);
    std::cout << hashMap.get(1) << std::endl;  // Output: 10
    std::cout << hashMap.get(3) << std::endl;  // Output: -1
    hashMap.put(2, 30);
    std::cout << hashMap.get(2) << std::endl;  // Output: 30
    hashMap.remove(2);
    std::cout << hashMap.get(2) << std::endl;  // Output: -1
    return 0;
}

File: JavaAnswer/Q1.java
import java.util.*;

class HashMap {
    private class Pair {
        int key;
        int value;
        
        Pair(int key, int value) {
            this.key = key;
            this.value = value;
        }
    }
    
    private List<List<Pair>> map;
    private int size;
    
    public HashMap(int size) {
        this.size = size;
        map = new ArrayList<>(size);
        for (int i = 0; i < size; i++) {
            map.add(new LinkedList<>());
        }
    }
    
    private int hash(int key) {
        return key % size;
    }
    
    public void put(int key, int value) {
        int hashValue = hash(key);
        List<Pair> bucket = map.get(hashValue);
        for (Pair pair : bucket) {
            if (pair.key == key) {
                pair.value = value;
                return;
            }
        }
        bucket.add(new Pair(key, value));
    }
    
    public int get(int key) {
        int hashValue = hash(key);
        List<Pair> bucket = map.get(hashValue);
        for (Pair pair : bucket) {
            if (pair.key == key) {
                return pair.value;
            }
        }
        return -1;
    }
    
    public void remove(int key) {
        int hashValue = hash(key);
        List<Pair> bucket = map.get(hashValue);
        bucket.removeIf(pair -> pair.key == key);
    }
    
    public static void main(String[] args) {
        HashMap hashMap = new HashMap(100);
        hashMap.put(1, 10);
        hashMap.put(2, 20);
        System.out.println(hashMap.get(1));  // Output: 10
        System.out.println(hashMap.get(3));  // Output: -1
        hashMap.put(2, 30);
        System.out.println(hashMap.get(2));  // Output: 30
        hashMap.remove(2);
        System.out.println(hashMap.get(2));  // Output: -1
    }
}
